### PR TITLE
Numbers

### PR DESCRIPTION
**Pull request by yulkadev**

### CODE REVIEW

File: Exercises/numbers/numbers.js [hunk 1]
```diff
@@ -40,9 +40,18 @@ console.log(randomFromZeroToOneHundred);
...
+}
 
-   function averageNumber() {
+averageNumber(myArray);
 
-   }
-//2. Generate array to pass to function above with values, and length generated by Math.random
\ No newline at end of file
+//2. Generate array to pass to function above with values, and length generated by Math.random
+let myArrayRandom = Array.from({length: 40}, () => Math.floor(Math.random() * 40));
```
Comment (Mike-Savin) [inline Exercises/numbers/numbers.js]: 👍
Finally somebody created this array not with loops but in a compact way.

File: Exercises/numbers/numbers.js [hunk 2]
```diff
@@ -40,9 +40,18 @@ console.log(randomFromZeroToOneHundred);
 //1. Make function which will take a array with numbers as a parameter and
    //calculate average number and round to hundredths.
 
-   let arrWithNumbers = [1, 5, 9, 0, 22, 2];
+var myArray = [0, 1, 2, 3, 4, 5, 444, 7, 8, 9, 10, 11];
+
+function averageNumber(myArray) {
+    var total = 0;
+    for (var i = 0; i < myArray.length; i++) {
```
Comment (Mike-Savin) [inline Exercises/numbers/numbers.js]: Use `let/const` instead of `var`

File: Exercises/numbers/numbers.js [hunk 3]
```diff
@@ -40,9 +40,18 @@ console.log(randomFromZeroToOneHundred);
...
+        total += myArray[i];
+    }
+        console.log(Math.round(total/myArray.length));
```
Comment (Mike-Savin) [inline Exercises/numbers/numbers.js]: Fix spaces and tabulation

Comment (Mike-Savin) [reply Exercises/numbers/numbers.js]: You need to not just round the final value but round it to two digits after comma.

File: Exercises/numbers/numbers.js [hunk 4]
```diff
@@ -40,9 +40,18 @@ console.log(randomFromZeroToOneHundred);
 //1. Make function which will take a array with numbers as a parameter and
    //calculate average number and round to hundredths.
 
-   let arrWithNumbers = [1, 5, 9, 0, 22, 2];
+var myArray = [0, 1, 2, 3, 4, 5, 444, 7, 8, 9, 10, 11];
```
Comment (Mike-Savin) [inline Exercises/numbers/numbers.js]: Use `let/const` instead of `var`

File: Exercises/numbers/numbers.js (hunk 1 above)
Comment (Mike-Savin) [reply Exercises/numbers/numbers.js]: Actually length also should be random but I don't think this matters

File: Exercises/numbers/numbers.js (hunk 2 above)
Comment (Mike-Savin) [reply Exercises/numbers/numbers.js]: You can also check a way of calculation of array sum using `reduce` method:
https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce

File: Exercises/numbers/numbers.js (hunk 3 above)
Comment (yulkadev) [reply Exercises/numbers/numbers.js]: Used reduce() to sum elements in array and .toFixed() to have only 2 digits after decimal point.

File: Exercises/numbers/numbers.js (hunk 4 above)
Comment (yulkadev) [reply Exercises/numbers/numbers.js]: Changed to let

File: Exercises/numbers/numbers.js [hunk 5]
```diff
@@ -1,3 +1,66 @@
+//1. Assign by hand same value to 3 variables in binary, octal and hexadecimal format.
+
+let binaryVar = 0b1000;
```
Comment (lizarusi) [inline Exercises/numbers/numbers.js]: always use `const` when variable not going to be change

File: Exercises/numbers/numbers.js [hunk 6]
```diff
@@ -1,3 +1,66 @@
...
+console.log(binaryVar); // =8
```
Comment (lizarusi) [inline Exercises/numbers/numbers.js]: the idea was to assign same value in different calculation systems but nevermind, you got the point

File: Exercises/numbers/numbers.js [hunk 7]
```diff
@@ -1,3 +1,66 @@
...
+let hexademicalVar = 0XA;
+console.log(hexademicalVar); // =10
+
+//2. Implement function `extractNumbers` which will leave only numbers in passed string.
+
 function extractNumbers (text) {
-  return null;
-}
\ No newline at end of file
+  let numbersOnly = text.replace(/\D+/g, '');
```
Comment (lizarusi) [inline Exercises/numbers/numbers.js]: 👍

File: Exercises/numbers/numbers.js [hunk 8]
```diff
@@ -1,3 +1,66 @@
...
+//calculate average number and round to hundredths.
+
+const myArray = [0, 1, 2, 3, 4, 5, 444, 7, 8, 9, 10, 11];
+
+function averageNumber(myArray) {
+  let total = 0;
+  for (let i = 0; i < myArray.length; i++) {
+    total += myArray[i];
+  }
+  console.log(Math.fround(total/myArray.length).toFixed(2));
```
Comment (lizarusi) [inline Exercises/numbers/numbers.js]: spaces around operator

Comment (lizarusi) [reply Exercises/numbers/numbers.js]: are you sure you need both  `fround` and `toFixed`?

File: Exercises/numbers/numbers.js [hunk 9]
```diff
@@ -1,3 +1,66 @@
...
+
+//# Additional
+
+//1. Make function which will take a array with numbers as a parameter and
+//calculate average number and round to hundredths.
+
+const myArray = [0, 1, 2, 3, 4, 5, 444, 7, 8, 9, 10, 11];
+
+function averageNumber(myArray) {
+  let total = 0;
```
Comment (lizarusi) [inline Exercises/numbers/numbers.js]: you need to collect statistic about each type

File: Exercises/numbers/numbers.js [hunk 10]
```diff
@@ -1,3 +1,66 @@
...
+  for (let i = 0; i < myArray.length; i++) {
```
Comment (lizarusi) [inline Exercises/numbers/numbers.js]: also very useful here is https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce

File: Exercises/numbers/numbers.js [hunk 11]
```diff
@@ -1,3 +1,66 @@
...
+  console.log(Math.fround(total/myArray.length).toFixed(2));
+}
+
+averageNumber(myArray);
+
+// Average number function with usage of reduce method
+
+function averageNumber2(myArray) {
+  let total = myArray.reduce((total, amount) => total + amount);
+  console.log((total/myArray.length).toFixed(2));  // "42.14"
```
Comment (lizarusi) [inline Exercises/numbers/numbers.js]: spaces

**Pull request merged**